### PR TITLE
Sync API for test logs

### PR DESCRIPTION
**Pull request by slimjim777**
Adding the sync API for testlogs:
 - use testlog record in factory instead of the filesystem
 - add the brand/model to the web form (so permissions can be determined in the cloud)
 - add testlog (factory and cloud)
 - list testlog (cloud)

### CODE REVIEW

File: datastore/testlogmanager.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	Filename string    `json:"filename"`
+	Data     string    `json:"data"`
+	Created  time.Time `json:"created"`
+	Synced   time.Time `json:"synced"` // used to indicate it has been synced to an external system
+}
+
+// CreateTestLogTable creates the database table for a test log
+func (db *DB) CreateTestLogTable() error {
+	_, err := db.Exec(createTestLogTableSQL)
+	if err != nil {
```
Comment (rmescandon) [inline datastore/testlogmanager.go]: return err

File: datastore/testlogmanager.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	}
+
+	return nil
+}
+
+// CreateTestLog keeps a record of a test log
+func (db *DB) CreateTestLog(testLog TestLog) error {
+	var err error
+	// Validate the data
+	if strings.TrimSpace(testLog.Brand) == "" || strings.TrimSpace(testLog.Model) == "" || strings.TrimSpace(testLog.Filename) == "" || strings.TrimSpace(testLog.Data) == "" {
```
Comment (rmescandon) [inline datastore/testlogmanager.go]: I wonder if a generic method to check and array of strings not to be empty... like notEmpty(args... string)

File: datastore/testlogmanager.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+		return errors.New("The brand, model, filename and file (base64-encoded) must be supplied")
+	}
+
+	// Create the signing log in the database
+	if Environ.Config.Driver == "sqlite3" {
```
Comment (rmescandon) [inline datastore/testlogmanager.go]: Uf.. this thing of having to deal with different drivers.. not in this pr, but in another maybe you find a kind of decorator  to have this thing for every needed query in a common place (not sure if that is possible either)

Comment (slimjim777) [reply datastore/testlogmanager.go]: I've toyed with the idea of putting the factory queries onto a separate interface, but the issue is that the factory uses the "standard" signing service. I'm not sure how to improve this at this stage. The database access has always felt a little messy.

**Pull request merged**